mlrun/datastore/base.py

```
@@ -157,6 +158,176 @@ def upload(self, key, src_path):
+                Example: [[('year','=',2025),('month','=',11),('timestamp','>',ts1)]]
+        :param partition_keys: (list of str) partition columns handled via directory
+
+        :return list of list of tuples: cleaned filters
+        """
+        cleaned_filters = []
+        for group in filters:
+            new_group = [f for f in group if f[0] not in partition_keys]
+            if new_group:
+                cleaned_filters.append(new_group)
+        return cleaned_filters
+
+    @staticmethod
+    def _read_partitioned_parquet(
+        base_path,
+        start_time,
+        end_time,
+        partition_keys,
+        df_module,
+        filesystem,
+        **kwargs,
+    ):
+        """Read only the relevant partitions using pandas filters and concat."""
+        logger.debug("starting paths partition process")
+        if not partition_keys:
+            raise ValueError(
+                f"No partition structure found under {base_path}, while usage requires partition keys"
+            )
+
+        paths = DataStore._list_partitioned_paths(
+            base_path,
+            start_time,
+            end_time,
+            partition_keys[-1],
+            filesystem,
+        )
+
+        dfs = []
+        errors = []
+        for current_path in paths:
+            try:
+                kwargs["filters"] = DataStore._clean_filters_for_partitions(
+                    kwargs["filters"], partition_keys
+                )
+                df = df_module.read_parquet(current_path, **kwargs)
+                logger.debug("Reading DataFrame", url=current_path, columns=df.columns)
+                dfs.append(df)
+            except (FileNotFoundError, pyarrow.lib.ArrowInvalid) as e:
+                # Skip partitions that don't exist or have no data
+                logger.warning(
+                    "Failed to read DataFrame", url=current_path, exception=e
+                )
+                errors.append(e)
+                continue
```

Don't exist yeah, but why are we suppressing real errors with `ArrowInvalid` and then raising the first error in the end anyway? Why not just raise it immediately? (by not catching it)

---

The way I worked before file could not exist, I will change it

---

We probably do want to tolerate a missing file, though. If only to avoid failure on concurrent deletions.

---

This one is still outstanding – I think it should tolerate missing partitions (with a warning for each seems reasonable), but not tolerate other failures (ArrowInvalid). Then, the warning should be specific to the directory being missing.
And we shouldn't be accumulating errors to raise one of them later. Fail early where we fail, and tolerate what we've decided to tolerate.